trace/noop.go

```diff
@@ -44,8 +44,8 @@ var _ Tracer = noopTracer{}
 
 // Start starts a noop span.
 func (t noopTracer) Start(ctx context.Context, name string, _ ...SpanOption) (context.Context, Span) {
-	span := noopSpan{}
-	return ContextWithSpan(ctx, span), span
+	// Don't set span in context since ContextFromSpan returns noopSpan{} by default.
+	return ctx, noopSpan{}
```

Comment:
This is certainly going to be faster, but is it correct?  What if there is an active, non-noop `Span` already in `ctx` when `Start()` is invoked?  Wouldn't that return a context with an active, non-noop `Span` alongside a noop `Span`?  In which case a later caller using `SpanFromContext(ctx)` might incorrectly modify or end the prior, non-noop `Span`.

Reply:
Is it possible to have both TracerProvider and NoopTracerProvider in the same app?

Reply:
It is possible and I don't believe that managing `TracerProvider`s is "unrealistic".  It is an intentional feature of the OTel API that multiple `TracerProvider`s may be used in an application.  I don't believe that we should be taking actions based on assumptions that certain combinations will not be used together.

Reply:
possible != realistic